src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1127,6 +1127,11 @@ mongoc_uri_apply_options (mongoc_uri_t *uri,
             }
 
             if (!mongoc_uri_set_option_as_bool (uri, canon, bval)) {
+               bson_set_error (error,
+                               MONGOC_ERROR_COMMAND,
+                               MONGOC_ERROR_COMMAND_INVALID_ARG,
+                               "Failed to set %s to %d",
+                               canon, bval);
```

Comment:
As it stands, I don't think we can test this. `mongoc_uri_set_option_as_bool` only fails if attempting to set an option that is not a boolean option or if it's overwriting an option that is stored as something other than a boolean. This is already inside an if-block checking that `mongoc_uri_option_is_bool`. I don't think either of those scenarios are possible, but still worthwhile to set an error if that changes in the future.